pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -40,13 +40,21 @@ type TemplateData struct {
 
 	// IronicUsername contains the password for authentication to Ironic
 	IronicPassword string
+
+	// APIVIP is the VIP to use for internal API communication
+	APIVIP string
+
+	// APIVIPv6 determines if our APIVIP is IPv6 or not
+	APIVIPv6 bool
 }
 
 // GetTemplateData returns platform-specific data for bootstrap templates.
 func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetworkEntry, ironicUsername, ironicPassword string) *TemplateData {
 	var templateData TemplateData
 
 	templateData.ProvisioningIP = config.BootstrapProvisioningIP
+	templateData.APIVIP = config.APIVIP
+	templateData.APIVIPv6 = net.ParseIP(config.APIVIP).To4() == nil
```

Comment:
What do you think about building `baremetal_endpoint_override` and `baremetal_introspection_endpoint_override` in go, instead of passing in the APIVIP to the template and bracketing there?


```golang
baremetalEndpointOverride := fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIP, "6385"))
baremetalIntrospectionEndpointOverride := fmt.Sprintf("http://%s/v1", net.JoinHostPort(config.APIVIP, "5050"))
```

This is an exceptionally contrived example, and exceedingly unlikely to happen in reality, however it's possible to have IPv6-mapped IPv4 addresses, and in those cases using `.To4()` to decide to bracket the original form is incorrect. See: https://play.golang.org/p/jQKo7GcU69H.

Reply:
I had not thought about that edge case. Building those two variables in go seems like a good workaround.